colpali_engine/trainer/contrastive_trainer.py

```diff
@@ -36,117 +25,12 @@ def prediction_step(self, model, inputs, prediction_loss_only, ignore_keys=True)
             raise ValueError("prediction_step is only called with prediction_loss_only=True")
 
         with torch.no_grad():
-            if self.is_vision_model:
-                if "doc_pixel_attention_mask" not in inputs:
-                    doc_outputs = model(
-                        input_ids=inputs["doc_input_ids"],
-                        attention_mask=inputs["doc_attention_mask"],
-                        pixel_values=inputs["doc_pixel_values"],
-                    )
-                else:
-                    doc_outputs = model(
-                        input_ids=inputs["doc_input_ids"],
-                        attention_mask=inputs["doc_attention_mask"],
-                        pixel_values=inputs["doc_pixel_values"],
-                        pixel_attention_mask=inputs["doc_pixel_attention_mask"],
-                    )
-                query_outputs = model(
-                    input_ids=inputs["query_input_ids"], attention_mask=inputs["query_attention_mask"]
-                )
-            else:
-                query_outputs = model(
-                    input_ids=inputs["query_input_ids"], attention_mask=inputs["query_attention_mask"]
-                )
-                doc_outputs = model(input_ids=inputs["doc_input_ids"], attention_mask=inputs["doc_attention_mask"])
+            doc_outputs = model(**{k[4:]: v for k, v in inputs.items() if k.startswith("doc")})
+            query_outputs = model(input_ids=inputs["query_input_ids"], attention_mask=inputs["query_attention_mask"])
+            if "neg_doc_input_ids" in inputs:
+                neg_doc_outputs = model(**{k[8:]: v for k, v in inputs.items() if k.startswith("neg_doc")})
+                loss = self.loss_func(query_outputs, doc_outputs, neg_doc_outputs)
+                return loss, None, None
 
             loss = self.loss_func(query_outputs, doc_outputs)
             return loss, None, None
-
-
-class ContrastiveNegativeTrainer(Trainer):
```

Comment:
If you're sure to remove `ContrastiveNegativeTrainer`, can add it to `## Removed` in `CHANGELOG` please?

Plus, I think this is **breaking** so maybe it'd be better to:
1. keep it for now and mark it as deprecated
2. OR release `0.4.0`

I'm more fan of method 1. Wdyt?

Reply:
This is just an explanation that's neededbecause it's completely non breaking as it's never an argument for training scripts. If people use it as a standalone, they can just use the normal one. We can just deprecate with a little message, it's best !
Important thing is not to break stuff at inference, here is fine

Reply:
Still removed it, worst case like 1 person gets annoyed but I think no one will